File: binary-search/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    if (nums.length === 0) return -1;

    const midpoint = Math.floor(nums.length / 2)

    if(nums[midpoint] === target){
        return midpoint
    } else if (nums[midpoint] > target){
        return search(nums.slice(0, midpoint), target)
    } else {
        const result = search(nums.slice(midpoint + 1), target)
        return result === -1 ? -1 : result + midpoint + 1
    }
};

File: leetcodepractice/792-binary-search/binary-search.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(nums, target)
    return -1 if nums.empty?

    mid = nums.length / 2
    comparator = nums[mid]

    if comparator > target
        return search(nums[0...mid], target)
    elsif comparator < target
        result = search(nums[mid + 1..-1], target)
        return result == -1 ? -1 : result + mid + 1
    else
        return mid
    end
end